sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala

```diff
@@ -1860,4 +1861,119 @@ class HiveDDLSuite
+        checkAnswer(
+          sql("SELECT * FROM tab WHERE c4 IS NULL"),
+          Seq(Row(1, 2, null))
+        )
+        assert(sql("SELECT * FROM tab").schema
+          .contains(StructField("c4", IntegerType)))
+        sql("INSERT INTO tab VALUES (2, 3, 4)")
+        checkAnswer(
+          sql("SELECT * FROM tab WHERE c4 = 4 "),
+          Seq(Row(2, 3, 4))
+        )
+        checkAnswer(
+          sql("SELECT * FROM tab"),
+          Seq(Row(1, 2, null), Row(2, 3, 4))
+        )
+      }
+    }
+  }
+
+  Seq("orc", "ORC", "org.apache.spark.sql.hive.orc",
+    "org.apache.spark.sql.hive.orc.DefaultSource").foreach { source =>
+    test(s"alter datasource table add columns - $source format not supported") {
+      withTable("tab") {
+        sql(s"CREATE TABLE tab (c1 int) USING $source")
+        val e = intercept[AnalysisException] {
+          sql("ALTER TABLE tab ADD COLUMNS (c2 int)")
+        }.getMessage
+        assert(
+          e.contains(s"ALTER ADD COLUMNS does not support datasource table with type"))
+      }
+    }
+  }
+
+  Seq("true", "false").foreach { caseSensitive =>
+    test(s"alter add columns with existing partition column name - caseSensitive $caseSensitive") {
+      withSQLConf(("spark.sql.caseSensitive", caseSensitive)) {
+        withTable("tab") {
+          sql("CREATE TABLE tab (c1 int) PARTITIONED BY (c2 int) STORED AS PARQUET")
+          if (caseSensitive == "false") {
+            val e = intercept[AnalysisException] {
+              sql("ALTER TABLE tab ADD COLUMNS (C2 string)")
+            }.getMessage
+            assert(e.contains("Found duplicate column(s)"))
+          } else {
+            // hive catalog will still complains that c1 is duplicate column name because hive
+            // identifiers are case insensitive.
+            val e = intercept[AnalysisException] {
+              sql("ALTER TABLE tab ADD COLUMNS (C2 string)")
+            }.getMessage
+            assert(e.contains("HiveException"))
+          }
+        }
+      }
+    }
+  }
+
+  Seq("true", "false").foreach { caseSensitive =>
+    test(s"alter add columns with existing column name - caseSensitive $caseSensitive") {
+      withSQLConf(("spark.sql.caseSensitive", caseSensitive)) {
+        withTable("t1") {
+          sql("CREATE TABLE t1 (c1 int) USING PARQUET")
+          if (caseSensitive == "false") {
+            val e = intercept[AnalysisException] {
+              sql("ALTER TABLE t1 ADD COLUMNS (C1 string)")
+            }.getMessage
+            assert(e.contains("Found duplicate column(s)"))
+          } else {
+            // hive catalog will still complains that c1 is duplicate column name because hive
+            // identifiers are case insensitive.
+            val e = intercept[AnalysisException] {
+              sql("ALTER TABLE t1 ADD COLUMNS (C1 string)")
+            }.getMessage
+            assert(e.contains("HiveException"))
+          }
+        }
+      }
+    }
+  }
```

Comment:
You still can combine it with the one in `InMemoryCatalogedDDLSuite` by using `isUsingHiveMetastore`